File: src/main/java/lana/thingService/attribute/AttributeServiceImpl.java
package lana.thingService.attribute;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.Optional;


@Service("attributeService")
public class AttributeServiceImpl implements AttributeService {
    private final AttributeRepo attributeRepo;

    @Autowired
    public AttributeServiceImpl(AttributeRepo attributeRepo) {
        this.attributeRepo = attributeRepo;
    }

    @Override
    public Page<Attribute> findAll(Pageable pageable) {
        return attributeRepo.findAll(pageable);
    }

    @Override
    public void delete(Integer id) {
        attributeRepo.deleteById(id);
    }

    @Override
    public Attribute create(Attribute attribute) {
        return attributeRepo.save(attribute);
    }

    @Override
    public Optional<Attribute> find(Integer id) {
        return attributeRepo.findById(id);
    }

    @Override
    public Optional<Attribute> update(Attribute attribute) {
        Optional<Attribute> existed = attributeRepo.findById(attribute.getId());
        if (existed.isPresent()) {
            existed = Optional.of(attributeRepo.save(attribute));
        }
        return existed;
    }
}

File: src/main/java/lana/thingService/attribute/AttributeServiceImpl.kt
package lana.thingService.attribute

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service("attributeService")
class AttributeServiceImpl
@Autowired
constructor(private val attributeRepo: AttributeRepo) : AttributeService {

    override fun findAll(pageable: Pageable): Page<Attribute> {
        return attributeRepo.findAll(pageable)
    }

    override fun find(id: Int): Attribute? {
        return attributeRepo.findById(id).get()
    }

    override fun delete(id: Int) {
        attributeRepo.deleteById(id)
    }

    override fun create(attribute: Attribute): Attribute? {
        return if (!isExist(attribute)) attributeRepo.save(attribute) else null
    }

    override fun update(attribute: Attribute): Attribute? {
        return if (isExist(attribute)) attributeRepo.save(attribute) else null
    }

    private fun isExist(attribute: Attribute): Boolean {
        val attributeId = attribute.id ?: return false
        return attributeRepo.existsById(attributeId)
    }
}
